src/components/DateInput/DateInput.tsx

```diff
@@ -102,16 +112,20 @@ const DateInput: React.FC<DateInputProps & Omit<TextInputProps, 'value' | 'onCha
     close();
   }, [close, value, timezone, setCurrentDate]);
 
+  const onClear = useCallback(() => setCurrentDate(undefined), [setCurrentDate]);
```

Comment:
To follow the convention above, I'd do 

```suggestion
  const onClear = useCallback(() => setCurrentDate(dateToDayjs(undefined)), [setCurrentDate]);
```